lib/parallel_tests/cucumber/scenarios.rb

```diff
@@ -10,22 +11,27 @@ module Cucumber
     class Scenarios
       class << self
         def all(files, options={})
-          tags = []
-          tags.concat options[:ignore_tag_pattern].to_s.split(/\s*,\s*/).map {|tag| "~#{tag}" }
-          tags.concat options[:test_options].to_s.scan(/(?:-t|--tags) (~?@[\w,~@]+)/).flatten
+          # Combine and generate a tag expression for given test options and ignore tag pattern. Refer here to understand how new tag expression syntax works - https://github.com/cucumber/cucumber/tree/master/tag-expressions
+          tags = ''
+          tags << options[:test_options].to_s unless options[:test_options].nil?
+          tags << " and " if !options[:ignore_tag_pattern].nil? && !options[:test_options].nil? # only add an 'and' operator if both test options and ignore tag pattern is passed
+          tags << "not (#{options[:ignore_tag_pattern].to_s})" unless options[:ignore_tag_pattern].nil?
 
```

Comment:
tags = [options[:test_options]]
if ignore = options[:ignore_tag_pattern]
  tags << "not (#{ignore})"
end
tags.compact.join(" and ")

Reply:
Alright, updated the code with one adjustment, I am doing `tags_exp = tags.compact.join(" and ")` because just `tags.compact.join(" and ")` won't update the `tags` variable from an array to a string.

```
tags = [options[:test_options]]
if ignore = options[:ignore_tag_pattern]
     tags << "not (#{ignore})"
end
tags_exp = tags.compact.join(" and ")

split_into_scenarios files, tags_exp
```